### ISSUE TITLE
Lock Screen Notifications settings don’t work

### ISSUE THREAD

**Issue by jmcken1** [thunderbird/thunderbird-android#3707]
### Expected behavior
Settings under Global settings → Notifications → Lock Screen Notifications should affect the content of lock screen notifications. _I.e._, “Message count and senders” should show the number of new emails and their senders, “Application name” should show only the K-9 Mail name and nothing else, and so on. (This presumes, of course, that the Android global setting to hide sensitive content in lock screen notifications is disabled.)

### Actual behavior
All settings under Global settings → Notifications → Lock Screen Notifications are completely ignored and have zero effect. Lock screen notifications are normal, revealing senders, subjects and body content. The only setting within K-9 Mail that affects lock screen notifications is Global settings → Privacy → Hide subject in notifications; when enabled, notifications show only the number of unread emails per account.

This happens both on my new Samsung Galaxy S8+ and my old Asus Zenfone 3 (ZE552KL).
All apps (including Android) are fully updated.

### Steps to reproduce
1. Go to Global settings → Notifications → Lock Screen Notifications and select any setting.
2. Lock phone and receive a new email (such as by sending one to yourself via another device).
3. Observe that chosen setting is completely ignored; senders, subjects and body content are all visible.
4. Optionally, enable Global settings → Privacy → Hide subject in notifications, which hides email content, but is still unaffected by the Lock Screen Notifications settings.

### Environment
K-9 Mail version: 5.600
Phone: Samsung Galaxy S8+
Android version: 8.0.0
Account type: IMAP (Gmail)

**Comment by jmcken1** [comment]
Am I the only one this happens to? Or was my bug report unclear? Would’ve expected someone to reply or close this or something by now. Just wondering.

**Comment by justdave** [comment]
I ran into this myself when I picked the "Hide subject" option...  which doesn't actually hide it only on the lock screen, but everywhere, if the message comes in while locked.  i.e. unlocking the phone doesn't return the information to the notification.  I forgot where I had set that so I was screwing with the Lock Screen Notifications setting in the meantime before I found it again and reproduced the above while playing with that. I'm currently running version 5.706

**Comment by cketti** [comment]
The *Hide subject* option didn't work particularly well (see comment by justdave). If we fixed it to work properly, it would most likely match the *Lock Screen Notifications* option *Message count and senders*. So with #5690 we got rid of the *Hide subject* setting.

I wasn't able to reproduce the issue of the *Lock Screen Notifications* setting not having any effect at all. However, it does require using a "secure" screen lock method (i.e. not "Swipe") and choosing the option "Show sensitive content only when unlocked" when setting up the screen lock in Android settings.

An issue I did run into was the lock screen notification being empty when there was just one new message. When there were multiple new messages the lock screen notification worked as intended.
The reason for that is Android's notification grouping mechanism not quite working as expected. When there's only one new message we're still treating it as a (notification) group. That means we create a notification for the individual message and a "summary" notification with essentially the same contents. On Android versions that don't support grouped notifications that summary notification is the only one that is displayed. On newer Android versions the notification sound and some other values are taken from the summary notification, but the collapsed notification is mostly generated from the contents of the individual notifications that make up the notification group. The "public" notification, i.e. the one that is displayed on secure lock screens, is a property of the summary notification. But only if the notification group consists of more than one message. If there's only one message, the "public" notification needs to be set on the individual notification (the notification sound is still taken from the summary notification :shrug:). This is quite annoying, but easy to work around.
